Data_Structures_Answers.md

```diff
@@ -1,18 +1,28 @@
 Add your answers to the questions below.
+O(1) describes an algorithm that will always execute in the same time (or space) regardless of the size of the input data set.
+O(N) describes an algorithm whose performance will grow linearly and in direct proportion to the size of the input data set.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
+O(1) using the array data structure
 
 2. What is the space complexity of your ring buffer's `append` function?
+and for the space it will be O(n) since we inserting in an array
 
 3. What is the runtime complexity of your ring buffer's `get` method?
+ O(n)
 
 4. What is the space complexity of your ring buffer's `get` method?
-
+O(n)
 
 5. What is the runtime complexity of the provided code in `names.py`?
+O(n)
```

Comment:
The runtime complexity of the provided code in names.py is O(n^2) due to the doubly-nested for loops that each iterate over the input list.